docs/compatible_microcontrollers.md

```diff
@@ -1,25 +1,34 @@
-# Atmel AVR
+# Compatible Microcontrollers
 
-QMK should run on any Atmel AVR processor with enough Flash. It has been tested on the following:
+## Atmel AVR
 
-* ATmega32U4 ([PJRC Teensy 2.0](http://www.pjrc.com/teensy/))
-* AT90USB1286 ([PJRC Teensy++ 2.0](http://www.pjrc.com/teensy/))
-* AT90USB1287 ([Atmel USBKEY](http://www.atmel.com/tools/AT90USBKEY.aspx))
-* ATmega168P with using [V-USB](http://www.obdev.at/products/vusb/index.html)
-* ATmega328P with using [V-USB](http://www.obdev.at/products/vusb/index.html)
-* ATmega32U2
-* AT90USB1286, 646, 647 should work
-* AT90USB162 testing...
+QMK runs on any Atmel USB AVR microcontroller with enough flash space - generally 32KB or more, though it will just squeeze into 16KB with most features disabled. The following use [LUFA](https://www.fourwalledcubicle.com/LUFA.php) as the USB stack:
 
-NOTE: To enable full features of firmware you'll need 32KB flash size.
+* [ATmega16U2](https://www.microchip.com/wwwproducts/en/ATmega16U2) / [ATmega32U2](https://www.microchip.com/wwwproducts/en/ATmega32U2)
+* [ATmega16U4](https://www.microchip.com/wwwproducts/en/ATmega16U4) / [ATmega32U4](https://www.microchip.com/wwwproducts/en/ATmega32U4)
+* [AT90USB64](https://www.microchip.com/wwwproducts/en/AT90USB646) / [AT90USB128](https://www.microchip.com/wwwproducts/en/AT90USB1286)
 
-Please add any tested microcontrollers to this list.
+Certain MCUs which do not have native USB will use [V-USB](https://www.obdev.at/products/vusb/index.html) instead:
 
-# ARM
+* [ATmega32A](https://www.microchip.com/wwwproducts/en/ATmega32A)
+* [ATmega328P](https://www.microchip.com/wwwproducts/en/ATmega328P)
 
-You can also use any ARM processor that [ChibiOS](http://www.chibios.org) supports. The following processors have been tested:
+## ARM
 
-* [Kinetis MKL26Z64](http://www.nxp.com/products/microcontrollers-and-processors/arm-processors/kinetis-cortex-m-mcus/l-series-ultra-low-power-m0-plus/kinetis-kl2x-48-mhz-usb-ultra-low-power-microcontrollers-mcus-based-on-arm-cortex-m0-plus-core:KL2x)
-* [Kinetis MK20DX128](http://www.nxp.com/assets/documents/data/en/data-sheets/K20P64M50SF0.pdf)
-* [Kinetis MK20DX128](http://www.nxp.com/assets/documents/data/en/data-sheets/K20P64M50SF0.pdf)
-* [Kinetis MK20DX256](http://www.nxp.com/products/microcontrollers-and-processors/arm-processors/kinetis-cortex-m-mcus/k-series-performance-m4/k2x-usb/kinetis-k20-72-mhz-full-speed-usb-mixed-signal-integration-microcontrollers-mcus-based-on-arm-cortex-m4-core:K20_72)
+You can also use any ARM chip that [ChibiOS](http://www.chibios.org) supports. These have plenty of flash. Known to work are:
```

Comment:
Not all ST chips have USB. Something like a f042 could run QMK, but the 16KB version will really struggle.
```suggestion
You can also use any ARM chip that [ChibiOS](http://www.chibios.org) supports that has USB. Most have plenty of flash. Known to work are:
```

Reply:
There are two F042 boards already in the repo, in fact. One of them calls out the STM32F042K6T6 specifically, which has 32KB.

Reply:
Different variants can have different amounts of flash, STM32F042K4 is 16KB.
STM32F030 runs chibios, but has no USB so kinda useless in the current QMK setup.

Reply:
Could maybe say that 32KB is the absolute minimum for STM32/ARM/both? Move the sentence that talks about it in the AVR section further up to an intro paragraph.